### PR TITLE
[IMP] hr_skill 11.0.2.0.0

### PR DESCRIPTION
**Pull request by osi-scampbell**
#503

### CODE REVIEW

File: hr_skill/models/hr_employee_skill.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class EmployeeSkill(models.Model):
+    _name = 'hr.employee.skill'
+
+    employee = fields.Many2one('hr.employee', string="Employee")
+
+    skill = fields.Many2one('hr.skill', string="Skill")
+    level = fields.Selection([('0', 'Junior'),
+                              ('1', 'Intermediate'),
+                              ('2', 'Senior'),
+                              ('3', 'Expert')], 'Level')
```
Comment (max3903) [inline hr_skill/models/hr_employee_skill.py]: `employee_id = `
`skill_id = `

newline at the end of the file

File: hr_skill/security/ir.model.access.csv [hunk 2]
```diff
@@ -1,3 +1,3 @@
 "id","name","model_id:id","group_id:id","perm_read","perm_write","perm_create","perm_unlink"
 "access_hr_skill","hr.skill","model_hr_skill","base.group_user",1,0,0,0
-"access_hr_skill_manager","hr.skill","model_hr_skill","hr.group_hr_manager",1,1,1,1
+"access_hr_skill_manager","hr.skill","model_hr_skill","hr.group_hr_manager",1,1,1,1
```
Comment (max3903) [inline hr_skill/security/ir.model.access.csv]: newline

File: hr_skill/models/hr_employee.py [hunk 3]
```diff
@@ -14,3 +14,5 @@ class Employee(models.Model):
         string='Skills',
         domain="[('child_ids', '=', False)]",
     )
+
+    employee_skill = fields.One2many('hr.employee.skill', 'employee', string="Employee Skill")
```
Comment (max3903) [inline hr_skill/models/hr_employee.py]: `employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Employee Skill") `

File: hr_skill/views/hr_employee.xml [hunk 4]
```diff
@@ -9,6 +9,7 @@
                 <page string="Skills">
                     <group>
                         <field name="skill_ids" widget="many2many_tags"/>
+                        <field name="employee_skill"/>
```
Comment (max3903) [inline hr_skill/views/hr_employee.xml]: `employee_skill_ids`

File: hr_skill/views/hr_employee_skill.xml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<odoo>
+  
+    <record id="view_employee_skill_form" model="ir.ui.view">
+        <field name="name">hr.employee.skill.form</field>
+        <field name="model">hr.employee.skill</field>
+        <field name="arch" type="xml">
+            <form string="Employee Skills">
+                <group>
+                    <field name="employee"/>
+                    <field name="skill"/>
```
Comment (max3903) [inline hr_skill/views/hr_employee_skill.xml]: `employee_id`
`skill_id`

**Review by max3903 — CHANGES_REQUESTED** [review]

File: hr_skill/security/ir.model.access.csv (hunk 2 above)
Comment (max3903) [reply hr_skill/security/ir.model.access.csv]: you also need to add the permissions for hr.employee.skill:

`"access_hr_employee_skill_user","hr.employee.skill","model_hr_employee_skill","hr.group_hr_user",1,1,1,1`

**Review by osi-scampbell — COMMENTED** [review]
Ok I have made those changes and pushed

File: hr_skill/security/ir.model.access.csv [hunk 6]
```diff
@@ -1,3 +1,4 @@
 "id","name","model_id:id","group_id:id","perm_read","perm_write","perm_create","perm_unlink"
 "access_hr_skill","hr.skill","model_hr_skill","base.group_user",1,0,0,0
 "access_hr_skill_manager","hr.skill","model_hr_skill","hr.group_hr_manager",1,1,1,1
+"access_employee_skill_user","hr.employee.skill","model_hr_employee_skill","hr.group_hr_user",1,1,1,1
```
Comment (max3903) [inline hr_skill/security/ir.model.access.csv]: Still missing the newline... :)

File: hr_skill/models/hr_employee.py [hunk 7]
```diff
@@ -14,3 +14,7 @@ class Employee(models.Model):
         string='Skills',
         domain="[('child_ids', '=', False)]",
     )
```
Comment (max3903) [inline hr_skill/models/hr_employee.py]: skill_ids should be removed.

File: hr_skill/views/hr_employee.xml [hunk 8]
```diff
@@ -9,6 +9,7 @@
                 <page string="Skills">
                     <group>
                         <field name="skill_ids" widget="many2many_tags"/>
```
Comment (max3903) [inline hr_skill/views/hr_employee.xml]: skill_ids should be removed.

**Comment by max3903** [comment]
@osi-scampbell We are still missing the pivot view and the migration script: #503

**Review by max3903 — APPROVED** [review]

**Review by smangukiya — APPROVED** [review]

**Pull request merged**